### PR TITLE
ci: add test for multiple os

### PR DESCRIPTION
**Pull request by smrpn**
Test project on multiple operating systems via matrix. 
ref - https://github.com/casbin/casbin-rs/issues/234

Signed-off-by: smrpn <samarpan_d@pp.iitr.ac.in>

### CODE REVIEW

**Comment by smrpn** [comment]
@hackerchai please review this.

**Comment by hackerchai** [comment]
@smrpn You need also to install depencies  for win/mac

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,8 +29,10 @@ jobs:
           components: rustfmt, clippy
           override: true
 
-      - name: Install Dependencies
-        run: sudo apt-get install libssl-dev
+      - name: Install Dependencies (for ubuntu)
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+            sudo apt-get install libssl-dev
```
Comment (hackerchai) [inline .github/workflows/ci.yml:35]: We also need macos & winodows to install dependency

Comment (smrpn) [reply .github/workflows/ci.yml:35]: Apparently, this dependency is not required. I ran the workflow without installing libssl-dev on ubuntu-latest, and it ran fine - https://github.com/smrpn/actix-casbin-auth/pull/1/checks?check_run_id=2426521309. 
Is it necessary to include that as a dependency?
Apart from that, libssl-dev in openssl in macOS which is preinstalled.

Comment (hackerchai) [reply .github/workflows/ci.yml:35]: how about the windows platform

Comment (smrpn) [reply .github/workflows/ci.yml:35]: Screenshot from 2021-04-25 13-47-27
The workflow ran on windows without installing the dependency.

Comment (hackerchai) [reply .github/workflows/ci.yml:35]: lgtm

**Pull request merged**